### PR TITLE
Add whats new page for 5.14 release

### PR DESCRIPTION
**Pull request by hasancelik**

### CODE REVIEW

File: docs/modules/ROOT/nav.adoc [hunk 1]
```diff
@@ -1,6 +1,7 @@
 .Get Started
 * xref:index.adoc[Overview]
 * xref:get-started.adoc[Deploy a Cluster]
+* xref:whats-new.adoc[What's New in 5.14]
```
Comment (oliverhowell) [inline docs/modules/ROOT/nav.adoc:4]:
```suggestion
* xref:whats-new.adoc[What's new in 5.14]
```

File: docs/modules/ROOT/pages/whats-new.adoc [hunk 2]
```diff
@@ -0,0 +1,28 @@
+= What's New in Hazelcast Platform Operator
```
Comment (oliverhowell) [inline docs/modules/ROOT/pages/whats-new.adoc:1]:
```suggestion
= What's new in Hazelcast Platform Operator
```

File: docs/modules/ROOT/pages/whats-new.adoc [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+:description: Here are the highlights of what's new and improved in Hazelcast Platform Operator 5.14.
+
+{description}
+
+
+== Lite Member Support
+
+With the release of Hazelcast Platform 5.5, Compute Isolation was introduced, allowing you to distribute your workload using Jet Job replacement control. To run workloads on Lite Members, the Platform Operator can enable Lite Member deployment through a Hazelcast custom resource. For more information xref:lite-members.adoc[Deploying Lite Members]
```
Comment (oliverhowell) [inline docs/modules/ROOT/pages/whats-new.adoc:9]:
```suggestion
Compute Isolation was introduced in Hazelcast Platform 5.5, enabling you to distribute your workload using Jet Job replacement control. To run workloads on Lite Members, Platform Operator can enable Lite Member deployment through a Hazelcast custom resource. For more information, see xref:lite-members.adoc[Deploying Lite Members].
```

File: docs/modules/ROOT/pages/whats-new.adoc [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+To learn more about Compute Isolation, see https://docs.hazelcast.com/hazelcast/latest/whats-new#distribute-your-workload-with-jet-job-placement-control[What's new in Hazelcast Platform 5.5].
+
+
+== Vector Collection Custom Resource
```
Comment (oliverhowell) [inline docs/modules/ROOT/pages/whats-new.adoc:14]:
```suggestion
== Vector Collection custom resource
```

File: docs/modules/ROOT/pages/whats-new.adoc [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+With the release of Hazelcast Platform 5.5, a new data structure called Vector Collection was introduced to enhance semantic search capabilities. The Platform Operator now includes a custom resource for creating and configuring Vector Collections in a declarative manner. For more details xref:vector-collection-configuration.adoc[Configuring Vector Collections].
```
Comment (oliverhowell) [inline docs/modules/ROOT/pages/whats-new.adoc:16]:
```suggestion
With the release of Hazelcast Platform 5.5, a new data structure called Vector Collection was introduced to enhance semantic search capabilities. Hazelcast Platform Operator now includes a custom resource for creating and configuring Vector Collections in a declarative manner. For more details, see xref:vector-collection-configuration.adoc[Configuring Vector Collections].
```

File: docs/modules/ROOT/pages/whats-new.adoc [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+== Additional Improvements
+
+For detailed release notes that include new features and enhancements, breaking changes, deprecations and other fixes, see xref:release-notes.adoc[5.14 Release Notes].
+
+To get started with Platform Operator, see xref:get-started.adoc[Deploy a Cluster with the Hazelcast Platform Operator for Kubernetes].
+
+== API Reference
+
+Please refer to the xref:api-ref.adoc[API Reference] for a complete list of available parameters for custom resources.
```
Comment (oliverhowell) [inline docs/modules/ROOT/pages/whats-new.adoc:28]:
```suggestion
For a complete list of available parameters for custom resources, see xref:api-ref.adoc[API Reference] .
```

**Review by oliverhowell — APPROVED** [review]
Minor suggestions, approving in advance

**Review by cheels — APPROVED** [review]

**Pull request merged**